brave/brave-core
# Add support for reserving tokens

Resolves https://github.com/brave/brave-browser/issues/8887

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

- Enable Rewards
- Ensure that database is migated to v27 (check meta table)
- Ensure that `reserved_at` defaults to `0` in migrated `unblinded_tokens` table
- Make a one-time contribution and verify that tip gets an appropriate `reserved_at` timestamp
- Make a one-time contribution and force it to fail all retries; verify that `reserved_at` field is cleared

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
